food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```diff
@@ -94,8 +109,14 @@ public String postTruckReview(@AuthenticationPrincipal User user, @PathVariable
 
     @Secured({"ROLE_OWNER"})
     @PostMapping("/truck/create")
-    public Truck createTruck(@AuthenticationPrincipal User u, @RequestBody CreateTruckParams data) {
-        return truckService.createTruck(u.getId(), data.truckName);
+    public Truck createTruck(@AuthenticationPrincipal User u, @RequestBody CreateTruckParams data){
+        Truck truck = truckService.createTruck(u.getId(), data.truckName);
+        try {
+            indexingService.initiateIndexing();
+        }catch (InterruptedException e){
+            return truck;
```

Comment:
Might not be a good idea to swallow this exception without at least logging it with `log.warn()`